js/source/geojson_source.js

```diff
@@ -41,28 +44,13 @@ module.exports = GeoJSONSource;
  * map.removeSource('some id');  // remove
  */
 function GeoJSONSource(options) {
-    options = options || {};
+    this.options = options = options || {};
```

Comment:
@lucaswoj Heh, I also just noticed that the `this.options = options` is actually duplicated a couple lines below, too.

Couple possible notes on removing this:
1. I noticed that `VectorTileSource` does keep ahold of its options and uses them again in `serialize` -- that actually does seem to me like an okay reason to hang on to it, but I could also see an argument for doing the serialization more explicitly... what's your preference?
2. Currently, down in `_updateData`, `this.options` gets used to load up the params that are sent over to the worker `{geojson|cluster}.parse`, allowing for those two (and potential other) source types to reuse the dirty-checking logic and such.  But anyway there are better ways to solve this.